#labels command,line,documentation
#Description of command line arguments

= NAME: =

        - RAP-Green v1.0 -

= SYNOPSIS: =

        java -jar RapGreen point jar command args

= OPTIONS: =

*-g* gene_tree_file
        The input gene tree file

*-s* species_tree_file
        The input species tree file

*-og* gene_tree_file
        The output tree file (annotated with duplications)

*-phyloxml* gene_tree_phyloxml_file
        The output tree file (annotated with duplications) in phyloXML format

*-os* species_tree_file
        The output species tree file (limited to gene tree species)

*-or* reconciled_tree_file
        The output reconciled tree file (consensus tree, with reductions and losses)

*-stats* gene_tree_file
        The output scoring statistic file

*-gt* gene_threshold
        The support threshold for gene tree branch collapse (optional, default 80.0)

*-st* species_threshold
        The length threshold for species tree branch collapse (optional, default 10.0)

*-pt* polymorphism_threshold
        The length depth threshold to deduce to polymorphism, allelism ... (optional, default 0.05)

*-k* k_level
        The k-level of the subtree-neighbor measure (optional, default 2)

*-idupw* i_duplication_weight
        The weight of intersection duplication in functional orthology scoring (0.0 for maximum weight, 1.0 for no weight, optional, default 0.90)

*-tdupw* t_duplication_weight
        The weight of topological duplication in functional orthology scoring (0.0 for maximum weight, 1.0 for no weight, optional, default 0.95)

*-specw* speciation_weight
        The weight of speciation in functional orthology scoring (0.0 for maximum weight, 1.0 for no weight, optional, default 0.99)

*-distw* distance_weight
        The weight of evolutionary distance in functional orthology scoring (0.0 for maximum weight, 1.0 for no weight, optional, default 0.10)