#PhyloXML output format example:

<?xml version="1.0" encoding="UTF-8"?>
<phyloxml xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.phyloxml.org http://www.phyloxml.org/1.10/phyloxml.xsd" xmlns="http://www.phyloxml.org">
<phylogeny rooted="true">
	<clade>
		<branch_length>0.0</branch_length>
		<confidence type="bootstrap">38.0</confidence>
		<events>
			<duplications>1</duplications>
		</events>
		<clade>
			<branch_length>0.05151904140499999</branch_length>
			<confidence type="bootstrap">38.0</confidence>
			<events>
				<duplications>1</duplications>
			</events>
			<clade>
				<branch_length>0.1438767603</branch_length>
				<confidence type="bootstrap">85.0</confidence>
				<events>
					<duplications>1</duplications>
				</events>
				<clade>
					<branch_length>0.0646807963</branch_length>
					<confidence type="bootstrap">85.0</confidence>
					<events>
						<speciations>1</speciations>
					</events>
					<clade>
						<branch_length>3.174E-7</branch_length>
						<confidence type="bootstrap">8.0</confidence>
						<events>
							<speciations>1</speciations>
						</events>
						<clade>
							<branch_length>1.0E-10</branch_length>
							<taxonomy>
								<code>ZEAMA</code>
							</taxonomy>
							<sequence id_source="GRMZM2G160041_P01">
								<name>GRMZM2G160041_P01</name>
							</sequence>
						</clade>
						<clade>
							<branch_length>3.174E-7</branch_length>
							<confidence type="bootstrap">11.0</confidence>
							<events>
								<speciations>1</speciations>
							</events>
							<clade>
								<branch_length>1.0E-10</branch_length>
								<taxonomy>
									<code>CLOUG</code>
								</taxonomy>
								<sequence id_source="GRMZM2G160041_P02">
									<name>GRMZM2G160041_P02</name>
								</sequence>
							</clade>
							<clade>
								<branch_length>8.92E-7</branch_length>
								<confidence type="bootstrap">19.0</confidence>
								<events>
									<speciations>1</speciations>
								</events>
								<clade>
									<branch_length>0.027020091</branch_length>
									<taxonomy>
										<code>ZEAMA</code>
									</taxonomy>
									<sequence id_source="GRMZM2G160041_P04">
										<name>GRMZM2G160041_P04</name>
									</sequence>
								</clade>
								<clade>
									<branch_length>1.0E-10</branch_length>
									<taxonomy>
										<code>ZEAMA</code>
									</taxonomy>
									<sequence id_source="GRMZM2G160041_P05">
										<name>GRMZM2G160041_P05</name>
									</sequence>
								</clade>
							</clade>
						</clade>
					</clade>
					<clade>
						<branch_length>1.0E-10</branch_length>
						<taxonomy>
							<code>ZEAMA</code>
						</taxonomy>
						<sequence id_source="GRMZM2G160041_P03">
							<name>GRMZM2G160041_P03</name>
						</sequence>
					</clade>
				</clade>
				<clade>
					<branch_length>0.0146121547</branch_length>
					<taxonomy>
						<code>SORBI</code>
					</taxonomy>
					<sequence id_source="Sb05g010340.1">
						<name>Sb05g010340.1</name>
					</sequence>
				</clade>
			</clade>
			<clade>
				<branch_length>0.0609865828</branch_length>
				<confidence type="bootstrap">77.0</confidence>
				<events>
					<speciations>1</speciations>
				</events>
				<clade>
					<branch_length>0.3786463422</branch_length>
					<taxonomy>
						<code>BRADI</code>
					</taxonomy>
					<sequence id_source="super_15.260_1">
						<name>super_15.260_1</name>
					</sequence>
				</clade>
				<clade>
					<branch_length>0.1577184955</branch_length>
					<taxonomy>
						<code>ORYZA</code>
					</taxonomy>
					<sequence id_source="Os11g29340.1">
						<name>Os11g29340.1</name>
					</sequence>
				</clade>
			</clade>
		</clade>
		<clade>
			<branch_length>0.002711528495</branch_length>
			<confidence type="bootstrap">38.0</confidence>
			<events>
				<speciations>1</speciations>
			</events>
			<clade>
				<branch_length>4.847079E-4</branch_length>
				<confidence type="bootstrap">14.0</confidence>
				<events>
					<speciations>1</speciations>
				</events>
				<clade>
					<branch_length>0.5557033691</branch_length>
					<confidence type="bootstrap">100.0</confidence>
					<events>
						<speciations>1</speciations>
					</events>
					<clade>
						<branch_length>0.0104364754</branch_length>
						<taxonomy>
							<code>SELMO</code>
						</taxonomy>
						<sequence id_source="Selmo_271942">
							<name>Selmo_271942</name>
						</sequence>
					</clade>
					<clade>
						<branch_length>1.0E-10</branch_length>
						<taxonomy>
							<code>SELMO</code>
						</taxonomy>
						<sequence id_source="Selmo_272257">
							<name>Selmo_272257</name>
						</sequence>
					</clade>
				</clade>
				<clade>
					<branch_length>0.2672115452</branch_length>
					<taxonomy>
						<code>ORYZA</code>
					</taxonomy>
					<sequence id_source="Os12g25640.1">
						<name>Os12g25640.1</name>
					</sequence>
				</clade>
			</clade>
			<clade>
				<branch_length>0.2588987711</branch_length>
				<confidence type="bootstrap">43.0</confidence>
				<events>
					<speciations>1</speciations>
				</events>
				<clade>
					<branch_length>0.0342883122</branch_length>
					<confidence type="bootstrap">16.0</confidence>
					<events>
						<speciations>1</speciations>
					</events>
					<clade>
						<branch_length>0.1004179346</branch_length>
						<taxonomy>
							<code>VITVI</code>
						</taxonomy>
						<sequence id_source="GSVIVP00017933001">
							<name>GSVIVP00017933001</name>
						</sequence>
					</clade>
					<clade>
						<branch_length>0.0677741096</branch_length>
						<confidence type="bootstrap">14.0</confidence>
						<events>
							<speciations>1</speciations>
						</events>
						<clade>
							<branch_length>0.0787852364</branch_length>
							<confidence type="bootstrap">89.0</confidence>
							<events>
								<speciations>1</speciations>
							</events>
							<clade>
								<branch_length>0.0111056987</branch_length>
								<confidence type="bootstrap">77.0</confidence>
								<events>
									<duplications>1</duplications>
								</events>
								<clade>
									<branch_length>0.0310578297</branch_length>
									<taxonomy>
										<code>POPAL</code>
									</taxonomy>
									<sequence id_source="Popal_666655">
										<name>Popal_666655</name>
									</sequence>
								</clade>
								<clade>
									<branch_length>0.0422684652</branch_length>
									<taxonomy>
										<code>POPAL</code>
									</taxonomy>
									<sequence id_source="Popal_423261">
										<name>Popal_423261</name>
									</sequence>
								</clade>
							</clade>
							<clade>
								<branch_length>0.0614035595</branch_length>
								<taxonomy>
									<code>RICCO</code>
								</taxonomy>
								<sequence id_source="30147.m014516">
									<name>30147.m014516</name>
								</sequence>
							</clade>
						</clade>
						<clade>
							<branch_length>0.0595254544</branch_length>
							<confidence type="bootstrap">33.0</confidence>
							<events>
								<speciations>1</speciations>
							</events>
							<clade>
								<branch_length>0.3057374926</branch_length>
								<taxonomy>
									<code>CARPA</code>
								</taxonomy>
								<sequence id_source="supercontig_13.117">
									<name>supercontig_13.117</name>
								</sequence>
							</clade>
							<clade>
								<branch_length>0.066324676</branch_length>
								<confidence type="bootstrap">77.0</confidence>
								<events>
									<speciations>1</speciations>
								</events>
								<clade>
									<branch_length>0.2106348403</branch_length>
									<taxonomy>
										<code>MEDTR</code>
									</taxonomy>
									<sequence id_source="AC174370_33.4">
										<name>AC174370_33.4</name>
									</sequence>
								</clade>
								<clade>
									<branch_length>0.0301772954</branch_length>
									<confidence type="bootstrap">80.0</confidence>
									<events>
										<duplications>1</duplications>
									</events>
									<clade>
										<branch_length>0.021854321</branch_length>
										<taxonomy>
											<code>SOYBN</code>
										</taxonomy>
										<sequence id_source="Glyma17g13860.1">
											<name>Glyma17g13860.1</name>
										</sequence>
									</clade>
									<clade>
										<branch_length>0.0427325437</branch_length>
										<taxonomy>
											<code>SOYBN</code>
										</taxonomy>
										<sequence id_source="Glyma05g03280.1">
											<name>Glyma05g03280.1</name>
										</sequence>
									</clade>
								</clade>
							</clade>
						</clade>
					</clade>
				</clade>
				<clade>
					<branch_length>0.1355484086</branch_length>
					<confidence type="bootstrap">50.0</confidence>
					<events>
						<speciations>1</speciations>
					</events>
					<clade>
						<branch_length>0.0587660922</branch_length>
						<confidence type="bootstrap">99.0</confidence>
						<events>
							<speciations>1</speciations>
						</events>
						<clade>
							<branch_length>1.0E-10</branch_length>
							<taxonomy>
								<code>ARATH</code>
							</taxonomy>
							<sequence id_source="At5g63460.1">
								<name>At5g63460.1</name>
							</sequence>
						</clade>
						<clade>
							<branch_length>1.795E-7</branch_length>
							<taxonomy>
								<code>ARATH</code>
							</taxonomy>
							<sequence id_source="At5g63460.2">
								<name>At5g63460.2</name>
							</sequence>
						</clade>
					</clade>
					<clade>
						<branch_length>0.4776284557</branch_length>
						<taxonomy>
							<code>PHYPA</code>
						</taxonomy>
						<sequence id_source="Phypa_27459">
							<name>Phypa_27459</name>
						</sequence>
					</clade>
				</clade>
			</clade>
		</clade>
	</clade>
<sequence_relation id_ref_0="GRMZM2G160041_P01" id_ref_1="GRMZM2G160041_P02" type="orthology">
	<confidence type="rap">0.99</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P01" id_ref_1="GRMZM2G160041_P04" type="orthology">
	<confidence type="rap">0.957</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P01" id_ref_1="GRMZM2G160041_P05" type="orthology">
	<confidence type="rap">0.98</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P01" id_ref_1="GRMZM2G160041_P03" type="orthology">
	<confidence type="rap">0.99</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P02" id_ref_1="GRMZM2G160041_P01" type="orthology">
	<confidence type="rap">0.99</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P02" id_ref_1="GRMZM2G160041_P04" type="orthology">
	<confidence type="rap">0.967</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P02" id_ref_1="GRMZM2G160041_P05" type="orthology">
	<confidence type="rap">0.99</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P02" id_ref_1="GRMZM2G160041_P03" type="orthology">
	<confidence type="rap">0.98</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P04" id_ref_1="GRMZM2G160041_P01" type="orthology">
	<confidence type="rap">0.957</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P04" id_ref_1="GRMZM2G160041_P02" type="orthology">
	<confidence type="rap">0.967</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P04" id_ref_1="GRMZM2G160041_P05" type="orthology">
	<confidence type="rap">0.976</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P04" id_ref_1="GRMZM2G160041_P03" type="orthology">
	<confidence type="rap">0.947</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P05" id_ref_1="GRMZM2G160041_P01" type="orthology">
	<confidence type="rap">0.98</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P05" id_ref_1="GRMZM2G160041_P02" type="orthology">
	<confidence type="rap">0.99</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P05" id_ref_1="GRMZM2G160041_P04" type="orthology">
	<confidence type="rap">0.976</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P05" id_ref_1="GRMZM2G160041_P03" type="orthology">
	<confidence type="rap">0.97</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P03" id_ref_1="GRMZM2G160041_P01" type="orthology">
	<confidence type="rap">0.99</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P03" id_ref_1="GRMZM2G160041_P02" type="orthology">
	<confidence type="rap">0.98</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P03" id_ref_1="GRMZM2G160041_P04" type="orthology">
	<confidence type="rap">0.947</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GRMZM2G160041_P03" id_ref_1="GRMZM2G160041_P05" type="orthology">
	<confidence type="rap">0.97</confidence>
</sequence_relation>
<sequence_relation id_ref_0="super_15.260_1" id_ref_1="Os11g29340.1" type="orthology">
	<confidence type="rap">0.539</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os11g29340.1" id_ref_1="super_15.260_1" type="orthology">
	<confidence type="rap">0.539</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.991</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.281</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.168</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.131</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_271942" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.135</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.991</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.29</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.176</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.131</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Selmo_272257" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.135</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.281</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.29</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.419</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.304</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.296</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.322</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.138</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.16</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.26</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.245</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.365</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.365</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Os12g25640.1" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.136</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.168</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.176</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.419</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.663</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.654</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.72</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.53</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.549</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.613</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.598</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.696</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.696</confidence>
</sequence_relation>
<sequence_relation id_ref_0="GSVIVP00017933001" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.35</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.304</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.663</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.819</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.513</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.53</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.584</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.57</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.549</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.549</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_666655" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.243</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.119</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.296</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.654</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.811</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.505</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.521</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.576</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.562</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.54</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.54</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Popal_423261" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.235</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.322</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.72</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.819</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.811</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.554</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.573</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.634</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.619</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.594</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.594</confidence>
</sequence_relation>
<sequence_relation id_ref_0="30147.m014516" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.254</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.132</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.138</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.53</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.513</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.505</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.554</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.494</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.566</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.55</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.41</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.41</confidence>
</sequence_relation>
<sequence_relation id_ref_0="supercontig_13.117" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.135</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.131</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.131</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.16</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.549</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.53</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.521</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.573</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.494</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.696</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.68</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.429</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.429</confidence>
</sequence_relation>
<sequence_relation id_ref_0="AC174370_33.4" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.26</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.613</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.584</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.576</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.634</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.566</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.696</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.502</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.502</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma17g13860.1" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.199</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.118</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.245</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.598</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.57</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.562</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.619</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.55</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.68</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.486</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.486</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Glyma05g03280.1" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.183</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.365</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.696</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.549</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.54</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.594</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.41</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.429</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.502</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.486</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">1.0</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.1" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.533</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.365</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.696</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.549</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.54</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.594</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.41</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.429</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.502</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.486</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">1.0</confidence>
</sequence_relation>
<sequence_relation id_ref_0="At5g63460.2" id_ref_1="Phypa_27459" type="orthology">
	<confidence type="rap">0.533</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Selmo_271942" type="orthology">
	<confidence type="rap">0.135</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Selmo_272257" type="orthology">
	<confidence type="rap">0.135</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Os12g25640.1" type="orthology">
	<confidence type="rap">0.136</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="GSVIVP00017933001" type="orthology">
	<confidence type="rap">0.35</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Popal_666655" type="orthology">
	<confidence type="rap">0.243</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Popal_423261" type="orthology">
	<confidence type="rap">0.235</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="30147.m014516" type="orthology">
	<confidence type="rap">0.254</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="supercontig_13.117" type="orthology">
	<confidence type="rap">0.135</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="AC174370_33.4" type="orthology">
	<confidence type="rap">0.134</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Glyma17g13860.1" type="orthology">
	<confidence type="rap">0.199</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="Glyma05g03280.1" type="orthology">
	<confidence type="rap">0.183</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="At5g63460.1" type="orthology">
	<confidence type="rap">0.533</confidence>
</sequence_relation>
<sequence_relation id_ref_0="Phypa_27459" id_ref_1="At5g63460.2" type="orthology">
	<confidence type="rap">0.533</confidence>
</sequence_relation>
</phylogeny>
</phyloxml>